expression/bench_test.go

```diff
@@ -416,6 +416,18 @@ func (g *randDurInt) gen() interface{} {
 	return int64(rand.Intn(types.TimeMaxHour)*10000 + rand.Intn(60)*100 + rand.Intn(60))
 }
 
+type randJsonDurationWithLen struct {
+	len int32
+}
+
+func (g *randJsonDurationWithLen) gen() interface{} {
+	j := new(json.BinaryJSON)
+	if err := j.UnmarshalJSON([]byte(fmt.Sprintf(`{"key":%v}`, rand.Int31n(g.len)))); err != nil {
```

Comment:
Why is it the form like `{key: <duration>}` but not `<duration>`? Does `builtinCastJSONAsDurationSig` have able to extract `<duration>` from the `key`?

Reply:
i think builtinCastJSONAsDurationSig indeed have able to extract duration from the JSON。The non vectorized implement like: 

func (b *builtinCastJSONAsDurationSig) evalDuration(row chunk.Row) (res types.Duration, isNull bool, err error) {
	val, isNull, err := b.args[0].EvalJSON(b.ctx, row)
	if isNull || err != nil {
		return res, isNull, err
	}
	s, err := val.Unquote()
	if err != nil {
		return res, false, err
	}
	res, err = types.ParseDuration(b.ctx.GetSessionVars().StmtCtx, s, int8(b.tp.Decimal))
	if types.ErrTruncatedWrongVal.Equal(err) {
		sc := b.ctx.GetSessionVars().StmtCtx
		err = sc.HandleTruncate(err)
	}
	return
}

Reply:
I don't think so.
After executing `val, isNull, err := b.args[0].EvalJSON(b.ctx, row)`, the `val` is `{"key": 11}` which is a JSON type.
After executing `val.Unquote()`, the `s` is `{"key": 11}` which is a string type.
Then types.ParseDuration parses `{"key": 11}` to a duration.

Reply:
I don't think so.
After executing `val, isNull, err := b.args[0].EvalJSON(b.ctx, row)`, the `val` is `{"key": 11}` which is a JSON type.
After executing `val.Unquote()`, the `s` is `{"key": 11}` which is a string type.
Then types.ParseDuration parses `{"key": 11}` to a duration.

Reply:
Sorry, after too long to reply. I had modified it. PTAL @SunRunAway @qw4990